src/binaries/query/main.rs

```diff
@@ -44,6 +44,7 @@ async fn main(_global_tracker: Arc<MemoryTracker>) -> common_exception::Result<(
         return Ok(());
     }
 
+    conf.meta.check_valid()?;
```

Comment:
Can we move this check to the config load?

Reply:
Or add an new function `check_valid` for `Config`. Inside the `check_valid` we can do our all valid checking.

Reply:
> Can we move this check to the config load?

If check when config loaded, some test may fail, cause `endpoints` and `embedded_dir` now has no default values.

Reply:
```
    let conf: Config = Config::load()?; -- Why test may fail if we check it in the end of load?

    if run_cmd(&conf) {
        return Ok(());
    }

    conf.meta.check_valid()?;
```

Reply:
Like this:
https://github.com/BohuTANG/databend/commit/04cda338e866c42cc50b3365bc8102ddee0b5875

I think it's OK.

Reply:
Or we can add a check_valid for Config.